src/main/java/duke/Duke.java

```diff
@@ -60,6 +60,15 @@ private static void respondToInput() {
                 } else if (checkIsInputEquals(userInput, "delete ")) {
                     int taskIndex = Integer.parseInt(userInput.substring("delete ".length())) - 1;
                     deleteAndDisplayTask(taskIndex);
+                } else if (checkIsInputEquals(userInput, "find ")) {
+                    String keyword = userInput.substring("find ".length());
+                    ArrayList<Task> matchingTasks = new ArrayList<>(100);
+                    for (Task task : tasks) {
+                        if (task.getInfo().contains(keyword)) {
+                            matchingTasks.add(task);
+                        }
+                    }
+                    displayList(matchingTasks);
                 } else {
                     dukeReply("I don't know what that means, sorry!");
```

Comment:
Specify where your error in this case is (invalid command at the start). Same for the rest of the exceptions in your code, so that it is clearer for the user to see what is wrong.